gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/handler/HttpBasicAuthenticationHandler.java

```diff
@@ -92,6 +102,13 @@ public void channelRead(final ChannelHandlerContext ctx, final Object msg) {
             try {
                 authenticator.authenticate(credentials);
                 ctx.fireChannelRead(request);
+
+                // User name logged with the remote socket address and authenticator classname for audit logging
+                if (authenticationSettings.enableAuditLog) {
+                    String[] authClassParts = authenticator.getClass().toString().split("[.]");
+                    auditLogger.info("User {} with address {} authenticated by {}", credentials.get(PROPERTY_USERNAME),
+                            ctx.channel().remoteAddress().toString().substring(1), authClassParts[authClassParts.length - 1]);
```

Comment:
Let me elaborate.  substring(1) assumes the toString() always starts with '/'.  However, if the hostname were resolved, then it would be in the format of "hostname/IP address:port".   substring(1) would result in "ostname/IP address:port".   It might not ever happen, but wanted to point it out for awareness.